scroll-search.lic

```diff
@@ -0,0 +1,533 @@
+=begin
+  Documentation: TBD
+=end
+
+custom_require.call(%w(common common-items drinfomon))
+
+# Define Scroll Info class
+class ScrollInfo
+  def initialize(container, index, container_index, ordinal, spell, type)
+	# Instance variables
+	@container = container
+	@index = index
+	@container_index = container_index
+	@ordinal = ordinal
+	@spell = spell
+	@type = type
+  end
+
+  def container
+	return @container
+  end
+
+  def spell
+	return @spell
+  end
+
+  def ordinal
+	return @ordinal
+  end
+  
+  def index
+	return @index
+  end
+  
+  # represents the local index of the
+  # scroll within the container
+  def container_index
+	return @container_index
+  end
+
+  def get_type
+	return @type
+  end
+
+  def set_container(container)
+	@container = container
+  end
+  
+  def set_container_index(index)
+	@container_index = index
+  end
+  
+  def set_ordinal(ordinal)
+	@ordinal = ordinal
+  end
+end
+
+class ScrollSearch
+  include DRC
+  include DRCI
+
+  def initialize
+
+	# Variables
+	@scroll_count = 0
+	@global_scroll_index = UserVars.scroll_counter.nil? ? 0 : UserVars.scroll_counter
+	@owned_scrolls = UserVars.owned_scrolls.empty? ? {} : UserVars.owned_scrolls
+    
+	# RESET ALL
+	# UserVars.owned_scrolls = { }
+	# @owned_scrolls = { }
+	# UserVars.scroll_counter = 0
+
+	$settings = get_settings()
+	$debug_mode = $settings.debug_mode
```

Comment:
This needs to have a unique name to avoid a collision with other scripts. For example, the debug flag in `addroom` is `debug_mode_ar`.